### PR TITLE
.github/workflows/test.yaml: fix logic

### PR DESCRIPTION
**Pull request by anonymouse64**
We want the spread tests to also run on pushes to release/** branches, so the
operator here should just be || otherwise these jobs will not run on pushes.
Thanks to Alberto for pointing this out to me.

Furthermore, we don't want to run the snap-builds job on anything other than
pull requests, since we already have Launchpad snap build recipes setup for the
release and master branches and this would just be duplicating that effort and
use up artifact storage space.

Also cleanup an unused setting that I had commented out as I wasn't sure if it 
would be necessary or not, but I can see now that it's not necessary.

Again, I'm deliberately not adding the skip spread label so we can see the actions 
run here just in case I have mucked things up again.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
Thanks, LGTM!

I'm afraid you have to close and reopen the PR, though, to let CI pick it up again.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -230,7 +231,7 @@ jobs:
 
   spread:
     needs: [unit-tests]
-    if: ${{ github.event_name != 'push' && github.ref != 'refs/heads/master' }}
+    if: ${{ github.event_name != 'push' || github.ref != 'refs/heads/master' }}
```
Comment (bboozzoo) [inline .github/workflows/test.yaml:234]: Thanks @mardy for spotting this.

Comment (bboozzoo) [inline .github/workflows/test.yaml:234]:
```suggestion
    # have spread jobs run on master on PRs only, but on both PRs and pushes to release branches
    if: ${{ github.event_name != 'push' || github.ref != 'refs/heads/master' }}
```

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by stolowski — APPROVED** [review]
LGTM (for my limited understanding of github actions ;))

File: .github/workflows/test.yaml (hunk 1 above)
Comment (anonymouse64) [reply .github/workflows/test.yaml:236]: applied, thanks

**Comment by anonymouse64** [comment]
> I'm afraid you have to close and reopen the PR, though, to let CI pick it up again.

@mardy actually you don't need to close to just re-run CI for changes like fixing the PR title, you can just click "re-run jobs" in the checks page. The only time you need to close and re-open a PR is for changing the labels, i.e. adding the "run nested" label or the "skip spread" labels.

**Pull request merged**